larray_editor/api.py

```diff
@@ -333,29 +370,31 @@ def excepthook(type, value, tback):
 
         main_tb = current_tb if _trace_code_file(current_tb) == main_file else tback
 
-        if usercode_traceback:
+        user_tb_length = None
+        if usercode_traceback or usercode_frame:
             if main_tb != current_tb:
                 print("Warning: couldn't find frame corresponding to user code, showing the full traceback "
                       "and inspect last frame instead (which might be in library code)",
                       file=sys.stderr)
-                limit = None
             else:
                 user_tb_length = 1
                 # continue as long as the next tb is still in the current project
                 while current_tb.tb_next and _trace_code_file(current_tb.tb_next).startswith(root_path):
                     current_tb = current_tb.tb_next
                     user_tb_length += 1
-                limit = user_tb_length
-        else:
-            limit = None
-        traceback.print_exception(type, value, main_tb, limit=limit)
+
+        tb_limit = user_tb_length if usercode_traceback else None
+        traceback.print_exception(type, value, main_tb, limit=tb_limit)
+
+        stack = extract_tb(main_tb, limit=tb_limit)
+        stack_pos = user_tb_length - 1 if user_tb_length is not None and usercode_frame else None
         print("\nlaunching larray editor to debug...", file=sys.stderr)
-        edit(_get_vars_from_frame(current_tb.tb_frame))
+        _debug(stack, stack_pos=stack_pos)
 
     return excepthook
 
 
-def run_editor_on_exception(root_path=None, usercode_traceback=True):
+def run_editor_on_exception(root_path=None, usercode_traceback=True, usercode_frame=True):
```

Comment:
Don't forget to repeat the new argument in the file ``viewer/__init__.pt`` in the ``larray`` repository

Reply:
will do